lib/srv/monitor.go

```diff
@@ -157,12 +157,12 @@ func (c *ConnectionMonitor) MonitorConn(ctx context.Context, authzCtx *authz.Con
 
 	tconn, ok := getTrackingReadConn(conn)
 	if !ok {
-		tctx, cancel := context.WithCancel(ctx)
+		tctx, cancel := context.WithCancelCause(ctx)
 		tconn, err = NewTrackingReadConn(TrackingReadConnConfig{
-			Conn:    conn,
-			Clock:   c.cfg.Clock,
-			Context: tctx,
-			Cancel:  cancel,
+			Conn:            conn,
+			Clock:           c.cfg.Clock,
+			Context:         tctx,
+			CancelWithCause: cancel,
```

Comment:
Can we replace Cancel with CancelWithCause instead of adding a new field?

Reply:
I was trying to avoid refactoring kube forwarder code. But let me give it a try.

Reply:
Removed the cancel now and updated kube session code.